tmail-backend/webadmin/webadmin-rate-limit/src/main/scala/com/linagora/tmail/webadmin/RateLimitPlanManagementRoutes.scala

```diff
@@ -0,0 +1,178 @@
+    service.get(GET_A_PLAN_PATH, getAPlan, jsonTransformer)
+    service.get(GET_ALL_PLAN_PATH, getAllPlan, jsonTransformer)
+  }
+
+  private def createAPlan: Route = (request: Request, response: Response) => {
+    SMono.fromPublisher(planRepository.create(toCreateRequest(request)))
+      .map(toRateLimitingPlanDTO)
+      .block()
+  }
+
+  private def updateAPlan: Route = (request: Request, response: Response) => {
+    SMono.fromPublisher(planRepository.update(toResetRequest(request)))
+      .onErrorResume { case e: RateLimitingPlanNotFoundException => SMono.error(
+        ErrorResponder.builder()
+        .statusCode(NOT_FOUND_404)
+        .`type`(ErrorResponder.ErrorType.NOT_FOUND)
+        .message("Plan does not exist")
+        .haltError())
+      }
+      .`then`(SMono.just(Responses.returnNoContent(response)))
+      .block()
+  }
+
+  private def getAPlan: Route = (request: Request, response: Response) => {
+    SMono.fromPublisher(planRepository.get(extractRateLimitingPlanId(request)))
+      .map(toRateLimitingPlanDTO)
+      .onErrorResume { case e: RateLimitingPlanNotFoundException => SMono.error(
+        ErrorResponder.builder()
+          .statusCode(NOT_FOUND_404)
+          .`type`(ErrorResponder.ErrorType.NOT_FOUND)
+          .message("Plan does not exist")
+          .haltError())
+      }
+      .block()
+  }
+
+  private def getAllPlan: Route = (request: Request, response: Response) => {
+    SFlux.fromPublisher(planRepository.list())
+      .map(toRateLimitingPlanDTO)
+      .collectSeq()
+      .map(seq => new GetAllRateLimitPlanResponseDTO(seq.toList.asJava))
+      .block()
+  }
+
+  private def toRateLimitingPlanDTO(rateLimitingPlan: RateLimitingPlan): RateLimitingPlanDTO = {
+    new RateLimitingPlanDTO(rateLimitingPlan.id.serialize(),
+      rateLimitingPlan.name.value,
+      toOperationLimitationsDTO(rateLimitingPlan.operationLimitations, OperationLimitations.TRANSIT_LIMITATIONS_NAME),
+      toOperationLimitationsDTO(rateLimitingPlan.operationLimitations, OperationLimitations.RELAY_LIMITATIONS_NAME),
+      toOperationLimitationsDTO(rateLimitingPlan.operationLimitations, OperationLimitations.DELIVERY_LIMITATIONS_NAME)
+    )
+  }
+
+  private def toOperationLimitationsDTO(operationLimitation: Seq[OperationLimitations], operationType: String): Optional[OperationLimitationsDTO] = {
+    operationLimitation.find(_.asString().equals(operationType))
+      .map(transitLimitation => new OperationLimitationsDTO(transitLimitation.rateLimitations().map(toRateLimitationDTO).toList.asJava))
+      .toJava
+  }
+
+  private def toRateLimitationDTO(rateLimitation: RateLimitation): RateLimitationDTO =
+    new RateLimitationDTO(rateLimitation.name, toPeriodString(rateLimitation.period),
+      extractLimitType(rateLimitation, LimitTypes.COUNT), extractLimitType(rateLimitation, LimitTypes.SIZE))
+
+  private def toPeriodString(duration: Duration): String = String.format("%s second", duration.getSeconds)
```

Comment:
I prefer durations to be expressed as a count of seconds (int) as here you would ask parsing from the client...